src/mpid/ch4/src/ch4_rma.h

```diff
@@ -14,20 +14,20 @@
 #include "ch4_impl.h"
 
 #undef FUNCNAME
-#define FUNCNAME MPID_Put
+#define FUNCNAME MPIDI_put_unsafe
 #undef FCNAME
 #define FCNAME MPL_QUOTE(FUNCNAME)
-MPL_STATIC_INLINE_PREFIX int MPID_Put(const void *origin_addr,
-                                      int origin_count,
-                                      MPI_Datatype origin_datatype,
-                                      int target_rank,
-                                      MPI_Aint target_disp,
-                                      int target_count, MPI_Datatype target_datatype,
-                                      MPIR_Win * win)
+MPL_STATIC_INLINE_PREFIX int MPIDI_put_unsafe(const void *origin_addr,
```

Comment:
Does this work without any changes in the synchronization functions?

Reply:
vni locks around sync functions were missing at that time. Now they are protected properly